### PR TITLE
Ben Fulton

### PR DESCRIPTION
**Pull request by b-enji-cmd**

### CODE REVIEW

**Comment by b-enji-cmd** [comment]
Damn, this one felt REALLY GOOD. Just couldn't make it work in time for only one book, if I had more time I am 100% confident I would have gotten all of iteration 3. Felt MUCH stronger on this than the last one. <3

**Comment by KathleenScriver** [comment]
Would love to see your TDD documented. So when you write the test, do a commit, then when the code is written to make that test pass, commit.

File: lib/book.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+        @title = info[:title]
+        @publication_date = info[:publication_date]
+    end
+
+    def author
+        @author_first_name + " " + @author_last_name
+    end
+
+    def publication_year
+        date = @publication_date.split(",").map(&:strip)[1]
```
Comment (KathleenScriver) [inline lib/book.rb:19]: Careful not to over-complicate this :)

File: lib/library.rb [hunk 2]
```diff
@@ -0,0 +1,48 @@
+class Library
+    attr_reader :name,
+                :books,
+                :authors
+
+    def initialize(name, books = [] , authors = [])
```
Comment (KathleenScriver) [inline lib/library.rb:6]: No need for default values in the arguments, we didn't ask for that.

File: lib/library.rb [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+    end
+    
+    def add_author(author)
+        @authors << author
+        @authors.map do |author|
+            author.books.each do |book|
+                @books << book
+            end
+        end
+        @authors
```
Comment (KathleenScriver) [inline lib/library.rb:19]: No need for this line. We didn't need to return anything specific from this method.

File: lib/library.rb [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+    end
+
+    def publication_time_frame_for(author)
```
Comment (KathleenScriver) [inline lib/library.rb:22]: Hopefully after the review you saw a less complicated way to do this. :) What you have here will work, but we can make it much more simple. This method though assumes that books are added in the order they were written, and that might not always be true.

**Review by KathleenScriver — COMMENTED** [review]
Good start here. I encourage you to reclone this challenge and try working through it again for repetition and practice. It would be a great exercise to go through it with your mentor as well. Your score is 2.5.

**Pull request closed without merging**